---
title: "indel_sequencing_summary"
output: html_document
date: "2024-03-10"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

bioc_pkgs <- c("tidyverse", "ggplot2", "viridis", "dplyr", "readxl", "stringr", "purrr", "gridExtra", "cowplot", "patchwork")
installBiocifnot <- function(pckgName){
  if (!(require(pckgName, character.only = TRUE))) {
    BiocManager::install(pckgName)
    require(pckgName, character.only = TRUE)
  }
}

for (i in 1:length(bioc_pkgs)){
  installBiocifnot(bioc_pkgs[i])
}

library(tidyverse)
library(ggplot2)
library(viridis)
library(dplyr)
library(readxl)
library(stringr)
library(purrr)
library(cowplot)
library(patchwork)

cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
sample_A1 <- read_excel("Plate1_abundance.xlsx", sheet = 1)
  sample_A1$Sample <- "AB8118"
  sample_A1$Genotype <- "cKO"
sample_A2 <- read_excel("Plate1_abundance.xlsx", sheet = 2)
  sample_A2$Sample <- "AB8119"
  sample_A2$Genotype <- "cKO"
sample_A3 <- read_excel("Plate1_abundance.xlsx", sheet = 3)
  sample_A3$Sample <- "AB8121"
  sample_A3$Genotype <- "CTRL"
sample_A4 <- read_excel("Plate1_abundance.xlsx", sheet = 4)
  sample_A4$Sample <- "AB8122"
  sample_A4$Genotype <- "CTRL"
indel_stats <- rbind(sample_A1, sample_A2, sample_A3, sample_A4)
indel_stats$Reads <- as.numeric(indel_stats$Reads)

df <- read_excel("genotype.xlsx")

```


```{r indel length histogram}
filtered_indel_stats <- na.omit(indel_stats)
indel_stats$Sample <- factor(indel_stats$Sample, levels = c("AB8118", "AB8121", "AB8119", "AB8122"))

# Create a bar plot showing the percentage of reads and overlay curves using ggplot2
hist_indel_length <- ggplot(data = indel_stats, aes(x = IndelLength, y = Pct, fill = Sample, color = Sample)) +
  geom_bar(stat = "identity", position = "identity", na.rm = TRUE) +
  xlim(-10, 10) +
  labs(
    y = "Percent of total reads",
    x = "Indel length (bp)",
    title = "Indel length distribution"
  )
hist_indel_length <- hist_indel_length + scale_colour_viridis_d() + scale_fill_viridis_d(alpha = 0.5) +
facet_wrap(~Sample + Genotype, scales = "fixed", ncol = 2) + 
theme(plot.title = element_text(hjust = 0.5)) 

hist_indel_length

ppi = 300
png(filename = "./plots/hist_indel_length.png", width = 8*ppi, height = 5*ppi, res = ppi)
  hist_indel_length
dev.off()

```


```{r indel frequency piechart}

df$Sample <- c("AB8118", "AB8119", "AB8121", "AB8122")
df$Genotype <- c("cKO", "cKO", "ctrl", "ctrl")
df$nonFrameshiftMutantReads <- df$MutantReads - df$FrameshiftMutantReads
df$WTReads <- df$TargetReads - df$MutantReads

create_pretty_pie_chart <- function(row) {
  values <- c(row$WTReads, row$FrameshiftMutantReads, row$nonFrameshiftMutantReads)
  labels <- c("WTReads", "FrameshiftMutantReads", "nonFrameshiftMutantReads")
  title <- paste(row$Sample, row$Genotype, sep="\n")
  colors <- c(cbPalette[2:4])  # Change to the desired colors
  
  total <- sum(values)
  percentages <- scales::percent(values / total)
  print(percentages)
  
  angle_midpoints <- cumsum(values) - 0.5 * values  # Calculate midpoints for each segment
  
  pie_chart <- ggplot() +
    geom_bar(stat = "identity", aes(x = "", y = values, fill = labels), width = 1, color = "white") +
    coord_polar("y", start = 0) +
    theme_void() +
    ggtitle(title) +
    theme(
      plot.title = element_text(hjust = 0.5, size = 20),  # Adjust the title size as needed
      legend.position = "none"  # Remove individual legends
    ) +
    scale_fill_manual(values = colors, name = "Categories")

   return(pie_chart)
}

# Create a list of pie charts
pie_chart_list <- lapply(1:nrow(df), function(i) create_pretty_pie_chart(df[i,]))

# Combine pie charts and add common legend using patchwork
common_legend <- get_legend(pie_chart_list[[1]])
pie_reads <- wrap_plots(pie_chart_list, ncol = 2) + plot_layout(guides = "collect") + theme(legend.position = "right", legend.text = element_text(size = 14), legend.title = element_text(size = 16, hjust = 0.5))

pie_reads

# Save the combined plot with a common legend as a PNG file
ggsave(filename = "./plots/read_category_piechart.png", 
       plot = pie_reads + common_legend,
       width = 10, height = 10, units = "in", dpi = 300, bg = "transparent")
```

```{r indel location histogram}

```


```{r}
session.info()
```

